packages/drivers/odsp-driver/src/odspDocumentServiceFactory.ts

```diff
@@ -25,13 +25,15 @@ export class OdspDocumentServiceFactory
         getWebsocketToken: TokenFetcher<OdspResourceTokenFetchOptions> | undefined,
         persistedCache?: IPersistedCache,
         hostPolicy?: HostStoragePolicy,
+        socketReferenceKeyPrefix?: string,
```

Comment:
I find it odd to have such semantics. I think either sockets should be agnostic of users (and support multiple logged in users per socket), or this argument should be removed / renamed to clearly indicate intention - it needs to uniquely identify the user.
I.e. be clear on intention - if scenario supports multiple users, then caller has to partition them by providing unique key here.

I'd think the better design would be for socket cache to be per factory, and simply state that factory supports only single user, allowing callers way out - usage of multiple factories.
That said, multi-user usage is not considered across layers. For example, cache (persistedCache agument) - is it Ok to use for multiple users? I'd rather state that it's a host problem to solve, and rely on same flow - factory per user, with host controlling fully how they construct (per-user) cache object. Alternatively this key needs to make it into cache interactions, in order to provide host ability to partition cache if they want to.

Reply:
I have considered changing the socket cache to per factory, but I think that will have negative effects on the current scenarios.  My goal here is to enable testing, so I don't want to do too major architecture change (to support multiple users).   I am sure there are other issues and usage that are not valid in the multiple user scenario.  That is the reason I choose the param name that are implementation specific, instead of usage specific (for now), until we fully design and support multiple user account usage of the driver in the same context.

Reply:
One possible alternative that I can do is to make this an option to have a per instance cache, versus a global cache so the caller can choose.  But the effect will be the same, and the flag will still be functional (what) specific instead of usage (why) specific.   The question is the work worth it at the moment.

Reply:
A cheap way to implement this is to generate a guid for each factory, and use that as the key prefix.  Is that sufficient?

Reply:
Yes, guid per factory is definitely sufficient, but you said it will have negative effects on the current scenarios?

Reply:
Only if we didn't give caller an option to switch it on and off.    Current in a lot of cases, developer create a driver per document.  If we always isolate the cache per factory, they will no longer able to share/reuse delta sockets.

I will change my PR to an `boolean` option and generate a guid if it is true.

Reply:
@vladsud  I made the change and put the optional flag in the options bag `ICollabSessionOptions`.

Reply:
I'm not sure it got better given lack of sharing between summarizer and non-summarizer client. We are definitely loosing some test coverage that way.

Reply:
Yep. Fixed by generating the uuid per factory